package/MDAnalysis/lib/formats/libmdaxdr.pyx

```diff
@@ -543,13 +511,99 @@ cdef class TRRFile(_XDRFile):
 
         if return_code == EOK:
             self.current_frame += 1
+        else:
+            raise IOError('TRR read error = {}'.format(
+                error_message[return_code]))
 
         has_x = bool(has_prop & HASX)
         has_v = bool(has_prop & HASV)
         has_f = bool(has_prop & HASF)
         return TRRFrame(xyz, velocity, forces, box, step, time, lmbda,
                         has_x, has_v, has_f)
 
+    def read_direct_xvf(self, np.float32_t[:, ::1] positions,
```

Comment:
can we remove `.read()` and in 3.0 make direct how we say that the XDRFile should be used, it's the much faster pattern

Reply:
Issue raised #3929